### ISSUE TITLE
crash while trying to load the avatar selection screen

### ISSUE THREAD

**Issue by mlc** [googlesamples/android-topeka#1]
I wrote a name and last initial, and then clicked the 'choose an avatar' button, and the app crashed:
```
FATAL EXCEPTION: main
Process: com.google.samples.apps.topeka, PID: 24314
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.samples.apps.topeka/com.google.samples.apps.topeka.activity.CategorySelectionActivity}: android.content.res.Resources$NotFoundException: File res/drawable/avatar_1.xml from drawable resource ID #0x7f020000
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2345)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2407)
    at android.app.ActivityThread.access$800(ActivityThread.java:149)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1324)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:211)
    at android.app.ActivityThread.main(ActivityThread.java:5321)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1016)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:811)
Caused by: android.content.res.Resources$NotFoundException: File res/drawable/avatar_1.xml from drawable resource ID #0x7f020000
    at android.content.res.Resources.loadDrawableForCookie(Resources.java:2648)
    at android.content.res.Resources.loadDrawable(Resources.java:2525)
    at android.content.res.Resources.getDrawable(Resources.java:799)
    at android.content.Context.getDrawable(Context.java:402)
    at com.google.samples.apps.topeka.activity.CategorySelectionActivity.setUpToolbar(CategorySelectionActivity.java:81)
    at com.google.samples.apps.topeka.activity.CategorySelectionActivity.onCreate(CategorySelectionActivity.java:59)
    at android.app.Activity.performCreate(Activity.java:5933)
    at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2298)
    ... 10 more
Caused by: java.lang.NumberFormatException: Invalid float: ".2.1"
    at java.lang.StringToReal.invalidReal(StringToReal.java:63)
    at java.lang.StringToReal.initialParse(StringToReal.java:160)
    at java.lang.StringToReal.parseFloat(StringToReal.java:323)
    at java.lang.Float.parseFloat(Float.java:306)
    at android.util.PathParser.getFloats(PathParser.java:178)
    at android.util.PathParser.createNodesFromPathData(PathParser.java:59)
    at android.graphics.drawable.VectorDrawable$VFullPath.updateStateFromTypedArray(VectorDrawable.java:1439)
    at android.graphics.drawable.VectorDrawable$VFullPath.inflate(VectorDrawable.java:1421)
    at android.graphics.drawable.VectorDrawable.inflateInternal(VectorDrawable.java:542)
    at android.graphics.drawable.VectorDrawable.inflate(VectorDrawable.java:458)
    at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:1126)
    at android.graphics.drawable.Drawable.createFromXml(Drawable.java:1033)
    at android.content.res.Resources.loadDrawableForCookie(Resources.java:2638)
    ... 18 more
```

I'm using a Sony Xperia Z3 Compact, with Android 5.0. Let me know if I can provide any more information.

**Comment by creativedrewy** [comment]
I think this is because the app is using xml resources that are generated into usable resources by build tools that haven't been officially released. I'm guessing to run the app we have to have the beta or canary versions of Android Studio/plugins.

Google, you might want to make this clear in the repo readme?

**Comment by robertoestivill** [comment]
Well, according `Getting Started` section of the `readme` file, you should be able to run it just by using one of the `Gradle` tasks. 

Also, all resources should be processed as part of the gradle build, not android studio. 
The Gradle build should fail if the configured version of the Android Gradle plugin is not found (meaning is not released). 

It sounds like just a bug to me 

`Caused by: java.lang.NumberFormatException: Invalid float: "-3.8.4"`

**Comment by mitchwongho** [comment]
Seeing the same on my OnePlus (stock CyanogenMod 5.0.2).  I've seen this type of error before when I've had a custom theme running on my device.

```
06-16 22:48:36.576  13055-13055/com.google.samples.apps.topeka E/PathParser﹕ error in parsing "l.2.1"
06-16 22:48:36.576  13055-13055/com.google.samples.apps.topeka D/AndroidRuntime﹕ Shutting down VM
06-16 22:48:36.577  13055-13055/com.google.samples.apps.topeka E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.google.samples.apps.topeka, PID: 13055
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.google.samples.apps.topeka/com.google.samples.apps.topeka.activity.CategorySelectionActivity}: android.content.res.Resources$NotFoundException: File res/drawable/avatar_1.xml from drawable resource ID #0x7f020000
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2329)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2389)
            at android.app.ActivityThread.access$900(ActivityThread.java:147)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
     Caused by: android.content.res.Resources$NotFoundException: File res/drawable/avatar_1.xml from drawable resource ID #0x7f020000
            at android.content.res.Resources.loadDrawableForCookie(Resources.java:2570)
            at android.content.res.Resources.loadDrawable(Resources.java:2472)
            at android.content.res.Resources.getDrawable(Resources.java:794)
            at android.content.res.Resources.getDrawable(Resources.java:770)
            at android.content.Context.getDrawable(Context.java:402)
            at com.google.samples.apps.topeka.activity.CategorySelectionActivity.setUpToolbar(CategorySelectionActivity.java:81)
            at com.google.samples.apps.topeka.activity.CategorySelectionActivity.onCreate(CategorySelectionActivity.java:59)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2282)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2389)
            at android.app.ActivityThread.access$900(ActivityThread.java:147)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:898)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:693)
     Caused by: java.lang.NumberFormatException: Invalid float: ".2.1"
            at java.lang.StringToReal.invalidReal(StringToReal.java:63)
            at java.lang.StringToReal.initialParse(StringToReal.java:160)
            at java.lang.StringToReal.parseFloat(StringToReal.java:323)
            at java.lang.Float.parseFloat(Float.java:306)
            at android.util.PathParser.getFloats(PathParser.java:178)
            at android.util.PathParser.createNodesFromPathData(PathParser.java:59)
            at android.graphics.drawable.VectorDrawable$VFullPath.updateStateFromTypedArray(VectorDrawable.java:1439)
            at android.graphics.drawable.VectorDrawable$VFullPath.inflate(VectorDrawable.java:1421)
            at android.graphics.drawable.VectorDrawable.inflateInternal(VectorDrawable.java:542)
            at android.graphics.drawable.VectorDrawable.inflate(VectorDrawable.java:458)
            at android.graphics.drawable.Drawable.createFromXmlInner(Drawable.java:1095)
            at android.graphics.drawable.Drawable.createFromXml(Drawable.java:1017)
            at android.content.res.Resources.loadDrawableForCookie(Resources.java:2560)
            at android.content.res.Resources.loadDrawable(Resources.java:2472)
            at android.content.res.Resources.getDrawable(Resources.java:794)
            at android.content.res.Resources.getDrawable(Resources.java:770)
            at android.content.Context.getDrawable(Context.java:402)
            at com.google.samples.apps.topeka.activity.CategorySelectionActivity.setUpToolbar(CategorySelectionActivity.java:81)
            at com.google.samples.apps.topeka.activity.CategorySelectionActivity.onCreate(CategorySelectionActivity.java:59)
            at android.app.Activity.performCreate(Activity.java:5933)
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2282)
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2389)
            at android.app.ActivityThread.access$900(ActivityThread.java:147)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1296)
            at android.os.Handler.dispatchMessage(Handler.java:102)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
```

**Comment by keyboardsurfer** [comment]
Fixed by 386edf455fc6cbcd92ffd0068a895aff2d6d4165

**Comment by mlc** [comment]
thanks for the quick fix, @keyboardsurfer!
